websites/C/ChompuBot/presence.ts

```diff
@@ -0,0 +1,146 @@
+	Dashboard = "/dashboard",
+	FAQ = "/faq",
+	Contact = "/contact",
+}
+
+let presenceData: PresenceData;
+
+function isPlayer(): boolean {
+	return !!document.querySelector<HTMLAnchorElement>(
+		"p.text-neutral-200.text-xl.text-center.mb-2"
+	);
+}
+
+presence.on("UpdateData", async () => {
+	const base = document.location.pathname;
+
+	if (/\/dashboard\/guild\/(.*[0-9_].*)\/music-room/gi.test(base)) {
+		let guildName,
+			title,
+			author,
+			playing,
+			pause,
+			timeStartPlayer,
+			timeEndPlayer,
+			StartPlayer,
+			durationPlayer,
+			startTimestamp,
+			endTimestamp;
+
+		if (!isPlayer()) {
+			guildName = document.querySelector<HTMLAnchorElement>(
+				"[data-label='player-requester']"
+			).textContent;
+			title = document.querySelector<HTMLAnchorElement>(
+				"h1.text-large.font-medium.-player-title"
+			);
+			author = document.querySelector<HTMLAnchorElement>(
+				"p.text-small.mt-1.text-foreground\\/80.-player-author"
+			);
+			playing = document.querySelector<HTMLAnchorElement>(
+				"svg.-player-playing"
+			);
+			pause = document.querySelector<HTMLAnchorElement>("svg.-player-pause");
+
+			(timeStartPlayer = document.querySelectorAll<HTMLElement>(
+				"p.text-small.-player-position-start"
+			)[0].textContent),
+				(timeEndPlayer = document.querySelectorAll<HTMLElement>(
+					"p.text-small.text-foreground\\/50.-player-position-end"
+				)[0].textContent),
+				([StartPlayer, durationPlayer] = [
+					presence.timestampFromFormat(timeStartPlayer),
+					(() => {
+						return presence.timestampFromFormat(timeEndPlayer);
+					})(),
+				]),
+				([startTimestamp, endTimestamp] = presence.getTimestamps(
+					StartPlayer,
+					durationPlayer
+				));
+		}
+
+		presenceData = {
+			details: "No song queue found",
+			state: "In the server...",
+			largeImageKey: document.querySelector<HTMLImageElement>(
+				"[data-label='guild-logo']"
+			).src,
+			smallImageKey: Assets.Reading,
+			smallImageText: "Zzz",
+			startTimestamp: browsingTimestamp,
+			buttons: [
+				{
+					label: "Join Player",
+					url: `https://chompubot.work${base}`,
+				},
+			],
+		};
+
+		if (title) {
+			presenceData.details = title;
+			presenceData.state = author;
+			presenceData.smallImageKey = playing ? Assets.Play : Assets.Pause;
+			presenceData.smallImageText = playing ? "Playing" : "Pause";
+			presenceData.startTimestamp = startTimestamp;
+			presenceData.endTimestamp = endTimestamp;
+			presenceData.buttons = [
+				{
+					label: `Join Player ${guildName}`,
+					url: `https://chompubot.work${base}`,
+				},
+			];
+			if (pause) {
+				delete presenceData.startTimestamp;
+				delete presenceData.endTimestamp;
+			}
+		}
+	} else {
+		presenceData = {
+			details: "Idk",
+			state: "Browsing...",
+			largeImageKey: Assets.Logo,
+			smallImageKey: Assets.Reading,
+			smallImageText: "Zzz",
+			startTimestamp: browsingTimestamp,
+			buttons: [
+				{
+					label: "Website",
+					url: "https://chompubot.work/",
+				},
+			],
```

Comment:
Buttons redirecting to main pages are not allowed by the [presence guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets).
```suggestion
```